python/pyfury/lib/meta/meta_string.py

```diff
@@ -0,0 +1,365 @@
+                return ord(c) - ord("a")
+            elif c == ".":
+                return 26
+            elif c == "_":
+                return 27
+            elif c == "$":
+                return 28
+            elif c == "|":
+                return 29
+            else:
+                raise ValueError(
+                    f"Unsupported character for LOWER_SPECIAL encoding: {c}"
+                )
+        elif bits_per_char == 6:
+            if "a" <= c <= "z":
+                return ord(c) - ord("a")
+            elif "A" <= c <= "Z":
+                return 26 + (ord(c) - ord("A"))
+            elif "0" <= c <= "9":
+                return 52 + (ord(c) - ord("0"))
+            elif c == self.special_char1:
+                return 62
+            elif c == self.special_char2:
+                return 63
+            else:
+                raise ValueError(
+                    f"Unsupported character for LOWER_UPPER_DIGIT_SPECIAL encoding: {c}"
+                )
+
+
+class MetaStringDecoder:
+    def __init__(self, special_char1, special_char2):
+        self.special_char1 = special_char1
+        self.special_char2 = special_char2
+
+    def decode(self, encoded_data, encoding):
+        if len(encoded_data) == 0:
+            return ""
+        return self.decode_with_encoding(encoded_data, encoding)
+
+    def decode_with_encoding(self, encoded_data, encoding):
+
+        if encoding == Encoding.LOWER_SPECIAL:
+            return self.decode_lower_special(encoded_data)
+        elif encoding == Encoding.LOWER_UPPER_DIGIT_SPECIAL:
+            return self.decode_lower_upper_digit_special(encoded_data)
+        elif encoding == Encoding.FIRST_TO_LOWER_SPECIAL:
+            return self.decode_rep_first_lower_special(encoded_data)
+        elif encoding == Encoding.ALL_TO_LOWER_SPECIAL:
+            return self.decode_rep_all_to_lower_special(encoded_data)
+        elif encoding == Encoding.UTF_8:
+            return encoded_data.tobytes().decode("utf-8")
+        else:
+            raise ValueError(f"Unexpected encoding flag: {encoding}")
+
+    def decode_lower_special(self, data):
+        decoded = []
+        num_bits = len(data) * 8
+        stripLastChar = (data[0] & 0x80) != 0
+        bit_index = 1
+        bit_mask = 0b11111
+        while bit_index + 5 <= num_bits and not (
+            stripLastChar and (bit_index + 2 * 5 > num_bits)
+        ):
+            byte_index = bit_index // 8
+            intra_byte_index = bit_index % 8
+            # // Extract the 5-bit character value across byte boundaries if needed
+            char_value = (
+                (data[byte_index] << 8)
+                | (data[byte_index + 1] if byte_index + 1 < len(data) else 0)
+            ) >> (11 - intra_byte_index) & bit_mask
+            bit_index += 5
+            decoded.append(self.decode_lower_special_char(char_value))
+
+        return "".join(decoded)
+
+    def decode_lower_upper_digit_special(self, data):
```

Comment:
```suggestion
    def _decode_lower_upper_digit_special(self, data):
```
private method should starts with `_`